sky/provision/aws/config.py

```diff
@@ -191,10 +191,9 @@ def _get_role(role_name: str):
             for policy_arn in attach_policy_arns:
                 role.attach_policy(PolicyArn=policy_arn)
 
-            # SkyPilot: 'PassRole' is required by the head node to pass the role
```

Comment:
Is the first part no longer accurate? Why?

Reply:
It is no longer accurate, as in our new provisioner, both the header and worker nodes are created with users' local credentials, instead of the credentials on the head node, so in multi-node cases, this permission is no longer needed.

However, in the case for jobs and services, the controller needs to still pass `skypilot-v1` role to the job cluster or service replicas created, and if the controller is with `remote_identity: SERVICE_ACCOUNT`, `skypilot-v1` needs to have those permissions.